datus/tools/func_tool/base.py

```
@@ -0,0 +1,62 @@
+    tool_template = function_tool(bound_method)
+
+    corrected_schema = json.loads(json.dumps(tool_template.params_json_schema))
+    if "self" in corrected_schema.get("properties", {}):
+        del corrected_schema["properties"]["self"]
+    if "self" in corrected_schema.get("required", []):
+        corrected_schema["required"].remove("self")
+
+    # The invoker MUST be an 'async' function.
+    # We define a closure to correctly capture the 'bound_method' for each iteration.
+    def create_async_invoker(method_to_call: Callable) -> Callable:
+        async def final_invoker(tool_ctx, args_str: str) -> dict:
```

Ruff (0.14.1)

Unused function argument: `tool_ctx`

(ARG001)